src/DynamicBody.cpp

```diff
@@ -371,7 +371,10 @@ bool DynamicBody::OnCollision(Body *o, Uint32 flags, double relVel)
 // return parameters for orbit of any body, gives both elliptic and hyperbolic trajectories
 Orbit DynamicBody::ComputeOrbit() const
 {
-	FrameId nrFrameId = Frame::GetFrame(GetFrame())->GetNonRotFrame();
+	auto f = Frame::GetFrame(GetFrame());
+	// if we are in a rotating frame, there can be no orbit
+	if (f->IsRotFrame()) return Orbit::ForStaticBody(GetPosition());
```

Comment:
If a ship is in a rotating frame it occurs to me that there definitely is an orbit (although for suborbital flight, the periapsis might fall below the surface of the body;) it is simply that the velocity of the ship is defined relative to the rotational velocity of the rotating frame. I'm not sure if our Orbit abstraction is robust enough to handle that case however.

If you could, I'd appreciate it if you could update the comment to mention the specific situation where getting the orbital parameters of a dynamic body currently under the influence of a rotational frame is not appropriate.

Reply:
Sure!

Reply:
Apologies, I was intending more that you should iterate *why* being in a rotating frame makes it inappropriate to retrieve orbital parameters for a body. Don't worry about it for now, I think this is good as is.